src/components/Toast/useToast.tsx

```diff
@@ -0,0 +1,18 @@
+import { useCallback } from "react";
+import { useToastContext } from "./ToastContext";
+import { ToastNoticeProps } from "./ToastNotice";
+
+export interface UseToastHook {
+  showToast: (props: ToastNoticeProps) => void;
+}
+export function useToast(): UseToastHook {
+  const { setNotice } = useToastContext();
+
+  const showToast = useCallback((props) => {
+    setNotice({
+      ...props,
+    });
+  }, []);
```

Comment:
Suggestion: Even though we "just know" that `setNotice` won't change, I think the react eslint rule will still want deps to be `[setNotice]`.